Source/WebCore/loader/SubresourceLoader.cpp

```diff
@@ -475,8 +475,25 @@ void SubresourceLoader::didReceiveResponse(const ResourceResponse& response, Com
         }
     }
 
-    if (m_resource)
-        m_resource->responseReceived(response);
+    if (m_loadingMultipartContent) {
+        if (!m_previousPartResponse.isNull()) {
+            if (m_resource) {
```

Comment:
```
if (CachedResourceHandle resource = m_resource.get()) {
    resource->responseReceived(m_previousPartResponse);
    // The resource data will change as the next part is loaded, so we need to make a copy.
    resource->finishLoading(resourceData()->copy().ptr(), { });
}
```

Since you're modifying this code, you might as well convert it to our safer / modern smart pointer idioms.